File: 백준/Silver/1251. 단어 나누기/단어 나누기.js
let fs = require('fs')
let input = fs.readFileSync('/dev/stdin').toString().trim().split("")

let answer = []
for (let i = 1; i < input.length; i++) {
    for (let j = i+1; j < input.length; j++ ) {
        let a = input.slice(0, i).reverse().join("")
        let b = input.slice(i, j).reverse().join("")
        let c = input.slice(j, input.length).reverse().join("")
        answer.push(a+b+c)        
    }
}
answer.sort()
console.log(answer[0])

File: 백준/Silver/1251. 단어 나누기/단어 나누기.py
import sys
input = sys.stdin.readline

s = input().rstrip()
s = list(s)
answer = []
for i in range(1, len(s)):
    for j in range(i+1, len(s)):
        a = s[:i]
        b = s[i:j]
        c = s[j:]

        a.reverse()
        b.reverse()
        c.reverse()

        answer.append("".join(a+b+c))

answer.sort()
print(answer[0])
